### PR TITLE
CATALOG-9783: [add] Product Images: V2 REST Contract

### PR DESCRIPTION
**Pull request by olysenko**
# [DEVDOCS-]

## What changed?
* Added Product Images: V2 REST Contract

## Release notes draft
*  Added Product Images: V2 REST Contract

### CODE REVIEW

File: archive/rest-catalog/product-images.v2.yml [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+    ProductImageWrite:
+      title: Writeable Image attributes
+      type: object
+      properties:
+        image_file:
+          type: string
+          description: |
+            When specifying a product image, the image_file should be specified as either:
+            a path to an image already uploaded via WebDAV to the import directory
+            (with the path relative to the import directory); or a URL to an image accessible on the internet.
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:260]:
```suggestion
-   (with the path relative to the import directory); or a URL to an image accessible on the internet

```

File: archive/rest-catalog/product-images.v2.yml [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+      description: 'Request body for product image update'
+    ProductImageWrite:
+      title: Writeable Image attributes
+      type: object
+      properties:
+        image_file:
+          type: string
+          description: |
+            When specifying a product image, the image_file should be specified as either:
+            a path to an image already uploaded via WebDAV to the import directory
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:259]:
```suggestion
-   a path to an image already uploaded via WebDAV to the import directory

```

File: archive/rest-catalog/product-images.v2.yml [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+          example: "sample_images/in_123__14581.jpg"
+        is_thumbnail:
+          type: boolean
+          description: If true, the image is used as the product's thumbnail.
+        sort_order:
+          maximum: 2147483647
+          minimum: -2147483648
+          type: integer
+          description: |
+            The order in which the image will be displayed on the product page. Higher integers give the image a lower priority.
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:270]:
```suggestion
            The order in which the image displays on the product page. Higher integers give the image a lower priority.
```

File: archive/rest-catalog/product-images.v2.yml [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+            True if it was setup from sample data. Otherwise is false.
+    ImagesCount:
+      title: Product Image Count
+      type: object
+      xml:
+        name: images
+      properties:
+        count:
+          type: integer
+          description:  A count of the number of Product Images in the store.
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:347]:
```suggestion
          description:  A count of the number of product images in the store.
```

File: archive/rest-catalog/product-images.v2.yml [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+          example: 100
+    error:
+      title: Product Image error response
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:350]:
```suggestion
      title: Product image error response
```

File: archive/rest-catalog/product-images.v2.yml [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+        'application/xml':
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/error'
+            xml:
+              name: errors
+              wrapped: true
+    ProductImage:
+      description: Get Product Image Response
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:379]:
```suggestion
      description: Get product image Response
```

File: archive/rest-catalog/product-images.v2.yml [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ProductImage'
+        application/xml:
+          schema:
+            $ref: '#/components/schemas/ProductImage'
+            xml:
+              name: images
+    ProductImages:
+      description: Gets array of Product Images Response
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:390]:
```suggestion
      description: Gets array of product images response
```

File: archive/rest-catalog/product-images.v2.yml [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+        'application/xml':
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/ProductImage'
+            xml:
+              name: images
+              wrapped: true
+    ProductImagesCount:
+      description: 'Gets a count of the total number of Product Images in the store.'
```
Comment (bc-traciporter) [inline archive/rest-catalog/product-images.v2.yml:406]:
```suggestion
      description: 'Gets a count of the total number of product images in the store.'
```

**Review by bc-traciporter — COMMENTED** [review]
I added editorial comments.

**Comment by becomevocal** [comment]
Closed this as after review we don't want to add more v2 Catalog API endpoint documentation since it's been deprecated for a while. We should be working to migrate to v3 and GraphQL at this point.

**Pull request closed without merging**